src/helpers/get-log.ts

```diff
@@ -14,17 +14,22 @@
  * app.log.fatal("Goodbye, cruel world!");
  * ```
  */
-import pino, { LoggerOptions } from "pino";
+import pino, { Logger, LoggerOptions } from "pino";
 import { getTransformStream, Options, LogLevel } from "@probot/pino";
+import { isProd } from "./is-prod";
 
 export type GetLogOptions = {
   level?: LogLevel;
   logMessageKey?: string;
 } & Options;
 
-export function getLog(options: GetLogOptions = {}) {
+export function getLog(options: GetLogOptions = {}): Logger {
   const { level, logMessageKey, ...getTransformStreamOptions } = options;
 
+  if (isProd() && !getTransformStreamOptions.logFormat) {
+    getTransformStreamOptions.logFormat = "json";
+  }
```

Comment:
This part isn't working when run from from the CLI because `logFormat` defaults to `"pretty"` when not otherwise defined:

https://github.com/probot/probot/blob/02a57d1bd2948ed019d5046c799f7e6fae770fc4/src/bin/read-cli-options.ts#L47

So `logFormat` is always defined, which means we never hit this block that sets it to json. We could either ignore `LOG_FORMAT` if `isProd()` or do something like:


```suggestion
  const logPretty = !isProd() && getTransformStreamOptions.logFormat == undefined
  getTransformStreamOptions.logFormat = logPretty ? "pretty" : "json";
```
The latter option replaces what we're doing in the CLI options, and should work just fine if you explicitly set `LOG_FORMAT` since it wouldn't be undefined when we get here. Thoughts?

Reply:
Thanks for your feedback, @tcbyrd. Two things to consider here:
- Should `json` and `pretty` be the only options? With your suggestion, we'd limit that.
- Should we always use `json` in production?

I'm curious what your thoughts are on this. Is there a guideline we should follow from probot's perspective?

My thinking was: If the user specifies _something_, we take it. If the user doesn't specify anything and we're in production, then we use `json`. This has the consequence that, when running from the CLI, the user would have to explicitly specify `json` if they want to log json.

Depending on what you think about the two points above, we could also change the default for the cli to something like `process.env.LOG_FORMAT || (isProd() ? "json" : "pretty")`.

Reply:
> Should json and pretty be the only options

`json` and `pretty` are the only formats built into pino. If you want another format, you'd have to use a custom pino logger. 

> ```js
> process.env.LOG_FORMAT || (isProd() ? "json" : "pretty").
> ```

Looks like the right approach to me.

Reply:
[addressed here](https://github.com/probot/probot/pull/1598/commits/32acbc62098d96a49cf2574d82a2ced8408504e9)